kubernetes-deployment/massdriver.yaml

```diff
@@ -46,50 +67,62 @@ params:
           type: integer
           default: 90
     ingress:
+      title: Ingress
+      description: Configure the application for public
       type: object
       properties:
         enabled:
+          title: Enable Public Internet Access
           type: boolean
         host:
+          title: Hostname
+          description: TBD, is this the right field name?
           type: string
         path:
           type: string
           default: '/'
-    envs:
-      type: array
-      items:
-        type: object
-        properties:
-          name:
-            type: string
-          value:
-            type: string
-<md if not .Connections md>
-connections:
-  properties: {}
-<md end -md>
 
-<md if .Connections md>
+# TODO: if this fails during template rendering because .Conditions isn't present, we shouldn't propogate nils, it should return an empty map
 connections:
   required:
+    - kubernetes_cluster
   <md- range $key, $art:= .Connections md>
     - <md $key md>
   <md- end md>
+  oneOf:
+    - required: ["aws_authentication"]
+    - required: ["gcp_authentication"]
+    - required: ["azure_authentication"]
   properties:
+    kubernetes_cluster:
+      $ref: massdriver/kubernetes-cluster
+    aws_authentication:
+      $ref: massdriver/aws-iam-role
+    gcp_authentication:
+      $ref: massdriver/gcp-service-account
+    azure_authentication:
+      $ref: massdriver/azure-service-principal
   <md- range $key, $art:= .Connections md>
     <md $key md>:
       $ref: <md $art md>
   <md- end md>
-<md- end md>
 
+artifacts:
+  properties:
+    debug:
+      $ref: massdriver/env-file
 ui:
+  image:
+    repository:
+      # Custom field docs: https://github.com/massdriver-cloud/docs/issues/8
+      ui:field: "containerRepositoriesDropdown"
+      # TODO: we need to make a cloud agnositc query var "any" or default to "any" if `cloud` is omitted
+      cloud: "aws"
```

Comment:
@xpositivityx we're going to need to support an all-the-clouds options. Maybe if not cloud is set, it defaults to listing all of your repos?

Reply:
I think this might be low prio in reality.

I think that our containerRepository dropdown will be useful in general purpose bundles (aws-lambda-pubsub-subscriber), but for peoples apps, it'll probably be hard coded.

Ie, massdriver platform API, will use the platform API, we wont change the image but will change the tag.